### PR TITLE
Initial support for ERC165

### PR DESCRIPTION
**Pull request by delaaxe**
Adds basic supports for ERC165 `supportsInterface` checking and custom interface registering.

### CODE REVIEW

**Review by Leonard-Pat — APPROVED** [review]

File: contracts/argent_account.cairo [hunk 1]
```diff
@@ -33,6 +34,26 @@ mod ArgentAccount {
     fn get_guardian_backup() -> felt {
         guardian_backup::read()
     }
+
+    // ERC165
+
+    #[view]
+    fn supportsInterface(interface_id: felt) -> bool {
+        // using combination of hardcoding and dynamic lookup for tradeoff between performance and flexibility
+        if interface_id == 0x01ffc9a7 {
```
Comment (gaetbout) [inline contracts/argent_account.cairo:43]: I'd use const rather then magic numbers if already possible

File: contracts/argent_account.cairo [hunk 2]
```diff
@@ -59,3 +80,22 @@ fn already_initialized() {
     assert(ArgentAccount::get_signer() == 1, 'value should be 1');
     ArgentAccount::initialize(10, 20, 0);
 }
+
+#[test]
+#[available_gas(20000)]
+fn erc165_basic_interfaces() {
```
Comment (gaetbout) [inline contracts/argent_account.cairo:86]: I'd split this in 2, to have the registered only on one side and an unregistered on the other side

File: contracts/argent_account.cairo (hunk 1 above)
Comment (delaaxe) [reply contracts/argent_account.cairo]: done

File: contracts/argent_account.cairo (hunk 2 above)
Comment (delaaxe) [reply contracts/argent_account.cairo]: done

**Comment by juniset** [comment]
Looking at the code, and given the uncertainty around e.g. naming conventions, I think we should stick to implementing what we have today in the account, and discuss potential improvements for later.

Can we remove the `supportedInterface` mapping for now and match the hardcoding we have in the Cairo 0.x version for now?

**Review by juniset — APPROVED** [review]

**Pull request merged**